src/Minigames/clash.js

```diff
@@ -0,0 +1,426 @@
+				AttackedES.clashAttackTimer = Date.now()
+				if (defense >= power) {
+					return void message.channel.send("Damn this guy's defenses sure are strong, you lost.")
+				}
+				const BrokeCell = AttackedMatrix.getCell(x, y)
+				switch (BrokeCell.value) {
+					case "M": EconomySystem.give(10000, message); break
+					case "T": {
+						let amount = 5000
+						for (const Cell of AttackedMatrix) {
+							if (Cell.value == "M") {
+								amount += 10000
+								Cell.value = "N"
+							}
+						}
+						EconomySystem.give(amount, message)
+						message.channel.send("Man you sure showed him who's boss huh?")
+					}
+				}
+				BrokeCell.value = "N"
+				message.channel.send("Congraturations you won the battle, *but not the war.*")
+			}
+			break
+        }
+        case "build": {
+            const EconomySystem = Economy.getEconomySystem(message.author);
+
+            if (EconomySystem.clashAttackTimer + Time.minute >= Date.now()) {
+            	return void message.channel.send("Dude you were just attacked, take a minute to rest and plan your violent revenge.");
+            }
+            if (!args[1]) {
+                message.channel.send("Uh-huh, I'll build your \" \" immediately.")
+                return
+            }
+            const ClashMatrix = EconomySystem.clash
+            const building = Clash.buildings[args[1].toLowerCase()]
+            if (!building) {
+                message.channel.send("I have no clue what a " + args[1] + " is.\nIf you have no clue what you can build either, try `&clash buildings`".replace(/\&/g, Prefix.get(message.guild.id)))
+            }
+            const total = ClashMatrix.getTotal(building)
+            if (total == 30) {message.channel.send("You already build 30 of these, you can't build more"); return}
+            const x = parseInt(args[2])
+            const y = parseInt(args[3])
+            if (isNaN(x) || isNaN(y)) {
+                message.channel.send("Now that position you just gave me...it doesn't make sense")
+                return
+            }
+            if (!ClashMatrix.checkBounds(x, y)) {
+                message.channel.send("Now that position you just told me...it's out of bounds")
+                return
+            }
+			if (!ClashMatrix.compare(x, y, "N")) {
+				return void message.channel.send("Dude you're gonna build on top of something else, move it first")
+			}
+            const price = 500 * EconomySystem.rank
+            if (EconomySystem.buy(price, message, "Sucessfully built your " + args[1], "You lack money for this building, you need " + price)) {
+                ClashMatrix.set(x, y, building)
+                if (building == "M") EconomySystem.clashtime = Date.now()
+            }
+            break
+        }
+        case "move": {
+            const xStart = parseInt(args[1])
+            const yStart = parseInt(args[2])
+            const xEnd = parseInt(args[3])
+            const yEnd = parseInt(args[4])
```

Comment:
`const [xStart, yStart, xEnd, yEnd] = args.slice(1, 5).map((arg) => parseInt(arg))` *might* be a shorter way to write this.

Reply:
aight I'll use that instead